src/include/distributed/multi_physical_planner.h

```diff
@@ -169,6 +169,9 @@ typedef struct Task
 	uint64 shardId;                /* only applies to shard fetch tasks */
 	TaskExecution *taskExecution;  /* used by task tracker executor */
 	bool upsertQuery;              /* only applies to modify tasks */
+
+	bool insertSelectQuery;
```

Comment:
>   The lock calls are idempotent within the same process, so it seems fine. 

Note that that's not quite true, because acquiring a lock multiple times
increases its lockcount, which is important when you explicitly release
the lock. But since we usually want locks till the end of xact anyway...